File: member_crud-jpa/app/src/main/java/sample/member/app/web/signup/SignupRequestDto.java
package sample.member.app.web.signup;

import lombok.Data;
import org.hibernate.validator.constraints.Length;

import javax.validation.constraints.AssertTrue;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotNull;

@Data
public class SignupRequestDto {

	@Length(min = 2, max = 10)
	@NotNull
	private String username;

	@Length(min = 2, max = 10)
	@NotNull
	private String realname;

	@Length(min = 5, max = 50)
	@NotNull
	@Email
	private String email;

	@Length(min = 1, max = 20)
	private String nickname;

	@Length(min = 5, max = 20)
	@NotNull
	private String password;

	@Length(min = 5, max = 20)
	@NotNull
	private String passwordConfirm;

	@AssertTrue
	public boolean isPasswordSame() {
		return password.equals(passwordConfirm);
	}
}

File: 2021-rabbitmq-jpa/app/src/main/kotlin/sample/app/web/signup/SignupRequestDto.kt
package sample.app.web.signup

import org.hibernate.validator.constraints.Length
import javax.validation.constraints.AssertTrue
import javax.validation.constraints.Email

data class SignupRequestDto(

    @field:Length(min = 2, max = 10)
//    @field:NotNull
    val username: String,

    @field:Length(min = 2, max = 10)
//    @field:NotNull
    val realname: String,

    @field:Length(min = 5, max = 50)
//    @field:NotNull
    @field:Email
    val email: String,

    @field:Length(min = 1, max = 20)
    val nickname: String?,

    @field:Length(min = 5, max = 20)
//    @field:NotNull
    val password: String,

    @field:Length(min = 5, max = 20)
//    @field:NotNull
    val passwordConfirm: String
) {
    @AssertTrue
    fun isPasswordEqual() = password == passwordConfirm
}
